cpp/ps0004/cpp_ps0004_solution.cpp

```diff
@@ -11,29 +12,26 @@ int main()
     int point = num;
     char ascii = num;
 
-    if (num == 0 || num == 'x')
+    // checking invalidaty of user's input 3/4
+	// printing the number and the opposite printable character 2/2
+    if (num == 0)
     {
         cout << "Invalid character/s or zero value.\n";
-
     }
     else if (num != point)
     {
         cout << "Invalid: a floating point number.\n";
-
     }
 
-    else if (num < 0 || num <= 32 && num >= 126)
+    else if (num < 32 || num > 126)
     {
         cout << "Invalid: number is out of the range.\n";
-
     }
     else
     {
         cout << "the character for  " << num << " is " << ascii << endl;
     }
+    
     return 0;
-
-
 }
-
-
+// good work 👍 7.5/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [x] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more